asistente_ladm_col/gui/dlg_get_java_path.py

```diff
@@ -0,0 +1,118 @@
+
+
+class DialogGetJavaPath(QDialog, DIALOG_UI):
+
+    def __init__(self, parent=None):
+        QDialog.__init__(self, parent)
+        self.parent = parent
+        self.setupUi(self)
+        self.setWindowTitle(QCoreApplication.translate("DialogGetJavaPath", "Get Java Path"))
+
+        self.java_path_line_edit.setPlaceholderText(
+            QCoreApplication.translate("DialogGetJavaPath", "[By default %PATH and %JAVA_HOME is searched]"))
+
+        self.java_path_search_button.clicked.connect(
+            make_file_selector(self.java_path_line_edit,
+                               QCoreApplication.translate("DialogGetJavaPath","Select Java application"),
+                               QCoreApplication.translate("DialogGetJavaPath","java (*)")))
+
+        self.java_path_line_edit.setValidator(FileValidator(is_executable=True, allow_empty=True))
+        self.validators = Validators()
+        self.java_path_line_edit.textChanged.connect(self.validators.validate_line_edits)
+
+        self.bar = QgsMessageBar()
+        self.bar.setSizePolicy(QSizePolicy.Minimum, QSizePolicy.Fixed)
+        self.layout().addWidget(self.bar, 0, 0, Qt.AlignTop)
+
+        self.buttonBox.accepted.disconnect()
+        self.buttonBox.accepted.connect(self.accepted)
+        self.buttonBox.clear()
+        self.buttonBox.addButton(QDialogButtonBox.Cancel)
+        self.buttonBox.addButton(QCoreApplication.translate("DialogGetJavaPath", "Set JAVA Path"),QDialogButtonBox.AcceptRole)
+
+    def accepted(self):
+        settings = QSettings()
+
+        (is_valid, java_message) = java_path_is_valid(self.java_path_line_edit.text().strip())
+
+        if is_valid:
+            self.java_path_line_edit.setEnabled(False)
+            settings.setValue('QgisModelBaker/ili2db/JavaPath', self.java_path_line_edit.text().strip())
+
+            self.buttonBox.clear()
+            self.buttonBox.setEnabled(True)
+            self.buttonBox.addButton(QDialogButtonBox.Close)
+            self.show_message(java_message, Qgis.Success)
+
+        else:
+            self.show_message(java_message, Qgis.Warning)
+            return
+
+    def show_message(self, message, level):
+        self.bar.pushMessage("Asistente LADM_COL", message, level, duration=0)
+
+
+def java_path_is_valid(java_path):
+    """
+    Check if java path exist
+    :param java_path: (str) java path to validate
+    :return: (bool, str)  True if java Path is valid, False in another case
+    """
+    try:
+        procs_message = subprocess.check_output([java_path, '-version'], stderr=subprocess.STDOUT).decode('utf8').lower()
+        types_java = ['jre', 'java', 'jdk']
+
+        if procs_message:
+            if any(type_java in procs_message for type_java in types_java):
+                pattern = '\"(\d+\.\d+).*\"'
+                java_version = re.search(pattern, procs_message).groups()[0]
+
+                if java_version:
+                    if float(java_version) >= JAVA_REQUIRED_VERSION:
```

Comment:
We require exactly 1.8. More recent versions break ili2db.

Reply:
Thank you. Change made